mqttclients/dotnet/MQTTnet.Client.Extensions/WithJWT.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Diagnostics;
+
+namespace MQTTnet.Client.Extensions
+{
+    public static partial class MqttNetExtensions
+    {
+        static Func<byte[]> _getTokenCallBack = null!;
+        static Timer _refreshTimer = null!;
+
+
+        public static MqttClientOptionsBuilder WithJWT(this MqttClientOptionsBuilder builder, MqttConnectionSettings cs, Func<byte[]> getTokenCallBack, IMqttClient mqttClient, TimeSpan refreshPeriod)
```

Comment:
@patilsnr this method is kind of weird, as it requires to pass the existing client to set the ReAuth flow with `SendExtendedAuthenticationExchangeDataAsync`

I think will be interesting to try to find a better flow, eg, by using the `IMqttExtendedAuthenticationExchangeHandler` from the client.Options.

thoughts?